service/kvmconfig/framework.go

```diff
@@ -144,22 +177,43 @@ func NewFramework(config FrameworkConfig) (*framework.Framework, error) {
 		}
 	}
 
-	// TODO route initCtx func together with resources.
-	initCtxFunc := func(ctx context.Context, obj interface{}) (context.Context, error) {
-		{
-			customObject, err := key.ToCustomObject(obj)
-			if err != nil {
-				return nil, microerror.Mask(err)
-			}
+	var v2ResourceSet *framework.ResourceSet
+	{
+		c := framework.ResourceSetConfig{}
 
-			versionBundleVersion := key.VersionBundleVersion(customObject)
+		c.Handles = v2ResourceSetHandles
+		c.Logger = config.Logger
+		c.Resources = v2Resources
 
-			if config.GuestUpdateEnabled && versionBundleVersion >= "1.1.0" {
-				updateallowedcontext.SetUpdateAllowed(ctx)
-			}
+		v2ResourceSet, err = framework.NewResourceSet(c)
+		if err != nil {
+			return nil, microerror.Mask(err)
 		}
+	}
 
-		return ctx, nil
+	var v3ResourceSet *framework.ResourceSet
+	{
+		c := framework.ResourceSetConfig{}
+
+		c.Handles = v3ResourceSetHandles
+		c.InitCtx = v3InitCtx
+		c.Logger = config.Logger
+		c.Resources = v3Resources
+	}
```

Comment:
I thought we can provide ResourceSets inside v3 -like packages. That would make imports easier.

Reply:
But it's ok for the time being. I can propose something later.

Reply:
There are probably further improvements we can make. I agree. Lets iterate from here.